gmp-task-utils-generated.h

```diff
@@ -0,0 +1,2416 @@
+/*
+ * Copyright 2015, Mozilla Foundation and contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// 0 arguments --
+template < typename M > class gmp_task_args_nm_0:public gmp_task_args_base {
+public:
+explicit gmp_task_args_nm_0 (M m):
+  m_ (m) {
+  }
+
+  void Run ()
+  {
+    m_ ();
+  }
+
+private:
+  M m_;
+};
+
+
+
+// 0 arguments --
+template < typename M, typename R > class gmp_task_args_nm_0_ret:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_nm_0_ret (M m, R * r):
+  m_ (m), r_ (r) {
```

Comment:
This should be imo intended more. Further following kind of constructor initialization list makes it cleaner when / if changes are done later. Personally I do not like those spaces after member variable either but that's very minor.
```
: m_(m)
, r_(r)
{}
```

Reply:
Ignore